### DISCUSSION TITLE
Does Tokio have a limit on the number of tasks

### DISCUSSION THREAD

**Discussion by xiashaohua** [tokio-rs/tokio#3649] [Q&A] [answered]
The code is:

```
    let topics = [TOPIC1,TOPIC2,TOPIC3];
    for &topic in topics.iter() {
        let message_sender_tmp = message_sender.clone();
        let  sessions_tmp = Arc::clone(&sessions);
        tokio::spawn(async move {
            sub_common_topic(sessions_tmp, message_sender_tmp, topic.to_string()).await;
        });
    }

    loop {
        tokio::select! {
            Some(res) = message_receiver.recv() => {
                let nc = nc.clone();
                println!("recv {:?}",res);
                tokio::spawn(async move{
                    println!("into task");
                    
                    let response = create_message_by_struct(res);
                    println!("step 1");
                    let topic = TOPIC_RESPONSE;
                    println!("step 2");
                    pub_to_topic_async(&nc, &topic, response.clone()).await;
                    info!("send message to topic: {},message is {:?}",topic.clone(),response);
                });
            }
            else => {
                continue
            }
        }
    }

```

when I add TOPIC4 to topics, the task in loop not run at all。If i add TOPIC4  and remove another topic. It will work  fine, I dont know why is it or is there something wrong with my usage.

**Comment by Darksonn** [comment, marked as answer]
No, there is no maximum number of tasks. If some of your tasks are not running, you should investigate if some of the other tasks are [blocking the thread][1], as this can prevent tasks from running.

One thing I notice is that the `continue` in your else branch would lead to an infinite loop. You only hit the else branch when the channel is closed, so you should break from the loop in that case.

[1]: https://ryhl.io/blog/async-what-is-blocking/

**Reply by xiashaohua** [reply]
I used the resident task to consume the topic,and use th sync api。The process should be blocked for this reason.The task works fine when I change to the asyn API

**Reply by xiashaohua** [reply]
thanks for your help
